runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -720,13 +721,21 @@ impl<T: Config> Pallet<T> {
 				      -> bool {
 					// never include a concluded-invalid candidate
 					concluded_invalid_disputes.contains(&backed_candidate.hash()) ||
-							// Instead of checking the candidates with code upgrades twice
-							// move the checking up here and skip it in the training wheels fallback.
-							// That way we avoid possible duplicate checks while assuring all
-							// backed candidates fine to pass on.
-							check_ctx
-								.verify_backed_candidate(parent_hash, candidate_idx, backed_candidate)
-								.is_err()
+						// Instead of checking the candidates with code upgrades twice
+						// move the checking up here and skip it in the training wheels fallback.
+						// That way we avoid possible duplicate checks while assuring all
+						// backed candidates fine to pass on.
+						{
+							match check_ctx.verify_backed_candidate(
+								parent_hash,
+								parent_storage_root,
+								candidate_idx,
+								backed_candidate,
+							) {
+								Err(_) | Ok(Err(_)) => true,
+								Ok(Ok(_)) => false,
```

Comment:
I'm confused why this doesn't match #4727, but this is what @drahnr suggested here: https://github.com/paritytech/polkadot/pull/4727/files#r793944509?